lib/ansible/module_utils/basic.py

```diff
@@ -1511,15 +1507,42 @@ def _handle_aliases(self, spec=None):
             for alias in aliases:
                 self._legal_inputs.append(alias)
                 aliases_results[alias] = k
-                if alias in self.params:
-                    self.params[k] = self.params[alias]
+                if alias in param:
+                    param[k] = param[alias]
 
         return aliases_results
 
-    def _check_arguments(self, check_invalid_arguments):
+    def _handle_no_log_values(self, spec=None, param=None):
```

Comment:
`_handle_no_log_values()` is also called from `_handle_options()` recursively in case on mullti-level argspec.
So if I understand your comment and referenced issue correctly,  this patch does resolve inheritance issue.